backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/impl/DocTypeBL.java

```diff
@@ -202,6 +202,50 @@ public boolean isDownPayment(@NonNull final DocTypeId docTypeId)
 				&& X_C_DocType.DOCBASETYPE_APInvoice.equals(dt.getDocBaseType());
 	}
 
+	@Override
+	public boolean isFinalInvoiceOrFinalCreditMemo(@NonNull final DocTypeId docTypeId)
+	{
+		return isFinalInvoice(docTypeId) || isFinalCreditMemo(docTypeId);
```

Comment:
u could load the doctype only once :)

Reply:
Did it like this so the function can be made public when this check alone is needed